bridgesample/src/androidTest/java/com/livefront/bridgesample/scenario/activity/LargeDataActivityTest.kt

```diff
@@ -0,0 +1,59 @@
+package com.livefront.bridgesample.scenario.activity
+
+import android.graphics.Bitmap
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.Espresso.pressBack
+import androidx.test.espresso.action.ViewActions.*
+import androidx.test.espresso.matcher.ViewMatchers.*
+import androidx.test.ext.junit.rules.ActivityScenarioRule
+import androidx.test.platform.app.InstrumentationRegistry
+import com.livefront.bridgesample.R
+import com.livefront.bridgesample.scenario.util.getCurrentActivity
+import org.junit.Assert.assertEquals
+import org.junit.Rule
+
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+class LargeDataActivityTest {
+
+    @get:Rule
+    var activityScenarioRule = ActivityScenarioRule<LargeDataActivity>(
+            LargeDataActivity.getNavigationIntent(
+                    InstrumentationRegistry.getInstrumentation().targetContext,
+                    false
+            )
+    )
+
+    @Test
+    fun generateDataAndNavigateNoCrash() {
+        onView(withId(R.id.generateDataButton)).perform(click())
+        onView(withId(R.id.navigateButton)).perform(click())
```

Comment:
So technically we can't be sure this is preventing a crash until we actually _have data_ right? Can we wait until we see the "Generated data..." message before we do the click?

Reply:
Same thing for the test below.

Also, can we add a test for rotation as well?

Reply:
My understanding is that Espresso will automatically wait for any kicked off AsyncTasks to complete before performing another action or assertion, so since the data is being generated in an AsyncTask I don't think we need to have add any additional logic.

Reply:
Hmm OK, good to know.